### PR TITLE
Fix/core pointed commit

### PR DESCRIPTION
**Pull request by mkondratek**
In scope of this PR we were up to limit computations of the pointed commit in `gitCore`. 

It turned out that the state of a `RevCommit` is "very" mutable. 
(`inDegree`, `flags` change during a rev walks)

Hence, `RevCommit` field has been removed from `GitCoreCommit` (to avoid potential bugs resulting from variable reuse).

---

**Note** that the aim of this task has been achieved (lazy evaluation of `getPointedCommit`).

### CODE REVIEW

File: gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java [hunk 1]
```diff
@@ -57,24 +65,29 @@ public String getName() {
   @SuppressWarnings("regexp")
   @Override
   public synchronized GitCoreCommit getPointedCommit() throws GitCoreException {
-    var gitCoreCommit = pointedCommitRef.get();
+    var gitCoreCommit = pointedCommit;
```
Comment (PawelLipski) [inline gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java:68]: I don't think this extra var is necessary

File: gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java [hunk 2]
```diff
@@ -57,24 +65,29 @@ public String getName() {
...
-      pointedCommitRef.set(value);
+      pointedCommit = value;
       return value;
     }
     return gitCoreCommit;
   }
 
   protected RevCommit derivePointedRevCommit() throws GitCoreException {
+    var revStr = getFullName();
+    return resolveRevCommit(revStr);
```
Comment (PawelLipski) [inline gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java:79]: Inline the var?

File: gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java [hunk 3]
```diff
@@ -24,22 +22,22 @@
...
 import com.virtuslab.gitcore.api.GitCoreException;
-import com.virtuslab.gitcore.api.GitCoreNoSuchBranchException;
 import com.virtuslab.gitcore.api.GitCoreNoSuchCommitException;
+import com.virtuslab.gitcore.api.GitCoreNoSuchRevisionException;
 import com.virtuslab.logger.IPrefixedLambdaLogger;
 import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
 
 @Getter
 @RequiredArgsConstructor
-@ToString
```
Comment (PawelLipski) [inline gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java:34]: Whoops... so basically no `toString` is provided now?...

**Comment by PawelLipski** [comment]
Btw pls retarget the PR to `develop`

File: gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java (hunk 3 above)
Comment (mkondratek) [reply gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java:34]: yup, we do not use it anyway

**Pull request merged**